### ISSUE TITLE
WASM build fails

### ISSUE THREAD

**Issue by kurorinchan** [Idein/msgpack-schema#52]
# tl;dr
WASM build fails, i.e. `cargo build --target=wasm32-unknown-unknown` fails.

Note that wasm build probably requires running `rustup target add wasm32-unknown-unknown` first.

# Problem

The build fails with the following error.

```
error: the wasm*-unknown-unknown targets are not supported by default, you may need to enable the "js" feature. For more information see: https://docs.rs/getrandom/#webassembly-support
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/getrandom-0.2.15/src/lib.rs:342:9
    |
342 | /         compile_error!("the wasm*-unknown-unknown targets are not supported by \
343 | |                         default, you may need to enable the \"js\" feature. \
344 | |                         For more information see: \
345 | |                         https://docs.rs/getrandom/#webassembly-support");
    | |________________________________________________________________________^

   Compiling libm v0.2.8
error[E0433]: failed to resolve: use of undeclared crate or module `imp`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/getrandom-0.2.15/src/lib.rs:398:9
    |
398 |         imp::getrandom_inner(dest)?;
    |         ^^^ use of undeclared crate or module `imp`
```

`getrandom` is a dependency of `proptest`. See it for yourself with `cargo tree`.

# Solutions
I am not very familiar with `proptest` nor the internals of this project.

Having that said I currently have several possible solutions.

## 1. Don't share workspace with msgpack-value
Remove [proptest as a default feature](https://github.com/Idein/msgpack-schema/blob/2ae4de745e96f0b9c2e0ceca352237fd30281392/Cargo.toml#L18). Also remove [msgpack-value from default members](https://github.com/Idein/msgpack-schema/blob/2ae4de745e96f0b9c2e0ceca352237fd30281392/Cargo.toml#L2). Both are required.

If feels like only the former is required because [default-features=false is used](https://github.com/Idein/msgpack-schema/blob/2ae4de745e96f0b9c2e0ceca352237fd30281392/Cargo.toml#L23). However since the workspace is shared with it, the [features are unified](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification). Which means the default feature `proptest` from `msgpack-value` is also used in the top level package, hence the flag is not doing anything.

The cons for this is that, if anyone is (for whatever reason) already relying on "proptest" feature as being the default of this library, then it would be a breaking change.

## 2. Remove `proptest` as defaults
Why is a "test" feature a default feature? Just remove it! IOW remove them from `Cargo.toml` in both the root and `msgpack-value`.
The cons of doing this is, the developer would have to run
`cargo build --all-features` or a similar command to verify all features build. `cargo test` should not be affected. Plus the same as above, changing default features could affect projects that depend on this.

## 3. Impose it on the user of the library
(I personally do not like this much because it requires user action, i.e. all projects that depend on this project.)

The user of the library could put 
`msgpack-schema = {version = "1.1.0", default-features = false }`
in their Cargo.toml. 

So add a note in README mentioning that "if you want WASM then put default-features=false".

**Comment by nyuichi** [comment]
Changing the default feature set requires a major version bump. It will not be short until we will release 2.0.0, as the crate itself is pretty stable.
For the moment please use the default-features=false switch, which is not an uncommon practice I think, though.
Anyway, we are happy to merge your pull request if you update the README to inform new users who want to use this crate with the wasm backend.
